src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,531 @@
+        /// </summary>
+        /// <returns></returns>
+        async Task IPartitionManager.StopAsync()
+        {
+            this.source.Cancel();
+            this.settings.Logger.PartitionManagerInfo(
+                "",
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "",
+                $"Worker {this.settings.WorkerId} starts shutting down.");
+            
+            await this.PartitionManagerStopLoop();
+
+            this.settings.Logger.PartitionManagerInfo(
+                "",
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "",
+                $"Worker {this.settings.WorkerId} finishes shutting down.");
+        }
+
+        async Task PartitionManagerStopLoop()
+        {
+            bool shouldRetry = false;
+            int timeToSleep = 1000;
+            
+            while (!shouldRetry)
+            {
+                try
+                {
+                    shouldRetry = await this.tableLeaseManager.ShutDown();
+                }
+                catch
+                {
+                    timeToSleep = 0;
+                }
+                await Task.Delay(timeToSleep);
+            }
+        }
+
+        async Task IPartitionManager.CreateLeaseStore()
+        {
+            string tableName = this.settings.AppName + "PartitionTable";
+            // will change later since managed identity can be different
+            var tableServiceClient = new TableServiceClient(this.settings.StorageConnectionString);
+            await tableServiceClient.CreateTableIfNotExistsAsync(tableName);
+            this.partitionTable = tableServiceClient.GetTableClient(tableName);
+        }
+
+        async Task IPartitionManager.CreateLease(string leaseName)
+        {
+            try
+            {
+                var lease = new TableLease()
+                {
+                    PartitionKey = "",
+                    RowKey = leaseName
+                };
+                await this.partitionTable.AddEntityAsync(lease);
+            }
+            catch(RequestFailedException)
+            {
+                //log info about that entity has already exists
+            }
+        }
+        
+        Task<IEnumerable<BlobLease>> IPartitionManager.GetOwnershipBlobLeases()
+        {
+            throw new NotImplementedException("This method is not implemented in the TablePartitionManager");
+        }
+
+        /// <summary>
+        /// internal use for testing . 
+        /// </summary>
+        /// <returns></returns>
+        internal IEnumerable<TableLease> GetTableLeases()
+        {
+            var partitions = this.partitionTable.Query<TableLease>();
+            return partitions;
+        }
+
+        internal void KillLoop()
+        {
+            this.source.Cancel();
+        }
+
+        async Task IPartitionManager.DeleteLeases()
+        {
+            await this.partitionTable.DeleteAsync();
```

Comment:
Since this is a one-line method implementation, I recommend we simplify it by removing `async` and replacing `await` wth `return`. That will make the code faster and simplify debugging if there are exceptions.
```suggestion
            return this.partitionTable.DeleteAsync();
```